tests/interop_client/main.hs

```diff
@@ -506,6 +516,83 @@ runClientStreamingTest opts = do
         RpcError err ->
           return (Left (show err))
 
+runServerStreamingTest :: Options -> IO (Either String ())
+runServerStreamingTest opts = do
+  let
+    responseSizes = [31415, 9, 2653, 58979]
+    req = def { _StreamingOutputCallRequest'responseParameters = map (\n -> def { _ResponseParameters'size = n }) responseSizes }
+    assertResponse (expectedSize, resp) =
+      case decodeMessage (L.toStrict resp) of
+        Left err -> Left ("proto decoder says: " ++ err)
+        Right msg -> case _StreamingOutputCallResponse'payload msg of
+                       Nothing -> Left "no payload"
+                       Just payload -> if _Payload'body payload == expectedBody
+                                          then Right ()
+                                          else Left "payload does not match"
+      where
+        expectedBody = B.replicate (fromIntegral expectedSize) 0
+
+    assertResponses resps
+      | length resps == length responseSizes =
+        case errors of
+          [] -> Right ()
+          (err : _) -> Left err
+      | otherwise = Left ("number of received messages mismatch: " ++ show (length resps))
+      where
+        assertions = map assertResponse (responseSizes `zip` resps)
+        errors = lefts assertions
+  bracket (newChannel opts) destroyChannel $ \channel ->
+    bracket (newClientContext channel) destroyClientContext $ \ctx -> do
+      client <- callDownstream ctx callOptions "/grpc.testing.TestService/StreamingOutputCall" (encodeMessage req)
+      resps <- withNewClient client $ do
+        msgs <- receiveAllMessages
+        closeCall
+        return msgs
+      case resps of
+        RpcOk resps' ->
+          return (assertResponses resps')
+        RpcError err ->
+          return (Left (show err))
+
+runServerStreamingWithSlowConsumerTest :: Options -> IO (Either String ())
```

Comment:
This test is not part of the interop testing suite.
It's also far too slow to have as a test here, it would make `--test_case=all` take very long.

But do feel free to move this to it's own test file, I propose
`tests/slow_consumer/main.hs`. Thanks!